test/core.jl

```diff
@@ -690,6 +690,18 @@ let A = [1]
     @test x == 1
 end
 
+# Pointer finalizer (issue #15408)
+let A = [1]
+    function f(to_finalize::Array{Int64,1})
+      to_finalize[1] = 0
+      nothing
+    end
+    cf = cfunction(f, Void, (Array{Int64,1},))
+    finalizer(A, cf)
```

Comment:
Passing a `cfunction` to `finalizer` is not allowed. It shouldn't cause a big issue in this case but it is better to use a real c function (possibly defined in `libccalltest`) instead.